t/10-jobs.t

```diff
@@ -418,7 +418,8 @@ subtest 'carry over, including soft-fails' => sub {
         is($inv->{test_log},    $fake_git_log, 'test git log is evaluated');
         is($inv->{needles_log}, $fake_git_log, 'needles git log is evaluated');
         $fake_git_log = '';
-        ok($inv = $job->investigate, 'job investigation ok for no test changes');
+        is($job->investigate->{test_log}, $inv->{test_log}, 'investigate caches per-job, git log not updated');
```

Comment:
As I said, it caches only per DBIx job object and I suppose a new of those objects is created for each request.

Reply:
I don't understand what you mean. This test shows that the hash is cached per server process so it prevents some redundant db retrieval. We already do the same for I think the settings hash.

Reply:
The whole test uses the same job object. So it only shows that it is cached for the lifetime of one object and not for the lifetime of the server process. Note that caching per server process wouldn't be a good idea anyways without any logic to get rid of old cache entries at some point.

The settings hash is only cached for the job object, too. But as far as I remember it makes sense here because the settings hash is often accessed multiple times per job object (from different functions so simply using a local variable when accessing it would be a bit complicated). If that caching wasn't per job object we would likely have a serious memory leak...

Reply:
> caching per server process wouldn't be a good idea anyways without any logic to get rid of old cache entries at some point.

isn't that done when either the server process restarts or the job is deleted?

Reply:
Even if per-process caching worked, with the number of processes we have the chance of hitting the one that has the job cached would be very small.